Apply this patch first:
	kdb-v4.4-3.1-common-1, kdb-v4.4-3.1-x86-1

then configure kernel:
     CONFIG_KDB=y, CONFIG_KGDB=y, CONFIG_KGDB_KDB=y
     CONFIG_DEBUG_INFO=y, CONFIG_DEBUG_RODATA=n

diff -urN linux-3.1_patch-kdb/arch/x86/include/asm/kdb.h linux-3.1_c-aries/arch/x86/include/asm/kdb.h
--- linux-3.1_patch-kdb/arch/x86/include/asm/kdb.h	2012-11-07 19:14:55.306652857 +0800
+++ linux-3.1_c-aries/arch/x86/include/asm/kdb.h	2012-11-07 19:06:32.785610234 +0800
@@ -39,7 +39,9 @@
 #define kdb_bfd_vma_fmt0	"0x%08lx"
 #define kdb_elfw_addr_fmt	"0x%x"
 #define kdb_elfw_addr_fmt0	"0x%08x"
+#ifndef kdb_f_count_fmt
 #define kdb_f_count_fmt		"%ld"
+#endif
 
 #else	/* CONFIG_X86_32 */
 
@@ -49,7 +51,9 @@
 #define kdb_bfd_vma_fmt0	"0x%016lx"
 #define kdb_elfw_addr_fmt	"0x%x"
 #define kdb_elfw_addr_fmt0	"0x%016x"
+#ifndef kdb_f_count_fmt
 #define kdb_f_count_fmt		"%ld"
+#endif
 
 /*
  * Functions to safely read and write kernel areas.  The {to,from}_xxx
diff -urN linux-3.1_patch-kdb/include/linux/kdb.h linux-3.1_c-aries/include/linux/kdb.h
--- linux-3.1_patch-kdb/include/linux/kdb.h	2012-11-07 19:14:48.236651046 +0800
+++ linux-3.1_c-aries/include/linux/kdb.h	2012-11-07 18:47:41.021737704 +0800
@@ -119,7 +119,10 @@
 	KDB_REASON_SILENT,	/* Silent entry/exit to kdb; regs invalid - internal only */
 } kdb_reason_t;
 
+#ifdef CONFIG_KDB
+#else
 #define kdb(reason,error_code,frame) (0)
+#endif
 
 #ifdef CONFIG_KGDB
 extern int kdb_trap_printk;
diff -urN linux-3.1_patch-kdb/kdb/kdbdereference.c linux-3.1_c-aries/kdb/kdbdereference.c
--- linux-3.1_patch-kdb/kdb/kdbdereference.c	2012-11-07 19:14:48.214651274 +0800
+++ linux-3.1_c-aries/kdb/kdbdereference.c	2012-11-07 19:07:52.911614088 +0800
@@ -5956,7 +5956,7 @@
 static int
 replace_unary(node_t *n, int flags)
 {
-	uint64_t value;
+	uint64_t value = 0;
 
 	if (!n->right) {
 		set_eval_error(E_MISSING_OPERAND);
diff -urN linux-3.1_patch-kdb/kernel/debug/kdb/kdb_private.h linux-3.1_c-aries/kernel/debug/kdb/kdb_private.h
--- linux-3.1_patch-kdb/kernel/debug/kdb/kdb_private.h	2011-10-24 15:10:05.000000000 +0800
+++ linux-3.1_c-aries/kernel/debug/kdb/kdb_private.h	2012-11-07 19:06:59.968609216 +0800
@@ -47,7 +47,9 @@
 #define kdb_bfd_vma_fmt0	"0x%08lx"
 #define kdb_elfw_addr_fmt	"0x%x"
 #define kdb_elfw_addr_fmt0	"0x%08x"
+#ifndef kdb_f_count_fmt
 #define kdb_f_count_fmt		"%d"
+#endif
 
 #elif BITS_PER_LONG == 64
 
@@ -59,7 +61,9 @@
 #define kdb_bfd_vma_fmt0	"0x%016lx"
 #define kdb_elfw_addr_fmt	"0x%x"
 #define kdb_elfw_addr_fmt0	"0x%016x"
+#ifndef kdb_f_count_fmt
 #define kdb_f_count_fmt		"%ld"
+#endif
 
 #endif
 
